### PR TITLE
Model interface

### PR DESCRIPTION
**Pull request by piojanu**
Add Modeler, Dynamics and Perfect Dynamics interfaces. Also refactor previous interfaces (Policy and Environment) to be abstract classes.

### CODE REVIEW

File: modeler.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        
+        Returns:
+          (object): Next state.
+          (float): Reward or final state value.
+          (bool): Is terminal.
+        """
+        pass
+
+
+class Modeler(metaclass=ABCMeta):
```
Comment (Matioz) [inline modeler.py]: typo? Modeler vs Modeller

File: modeler.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    def get_dynamics():
+        """Passes world dynamics used to simulate transitions.
+        
+        Return:
+          (Dynamics): World dynamics.
+        """
+        pass
+
+    @abstractmethod
+    def report_step(self, transition):
```
Comment (Matioz) [inline modeler.py]: I think but this method should be moved to separated interface named Callback

File: modeler.py (hunk 1 above)
Comment (piojanu) [reply modeler.py]: Nope: https://www.diki.pl/slownik-angielskiego?q=modeler

Comment (piojanu) [reply modeler.py]: Buy I agree it could have different name. Any ideas?

File: modeler.py (hunk 2 above)
Comment (piojanu) [reply modeler.py]: Will you do this?

Comment (Matioz) [reply modeler.py]: yes, I can

File: modeler.py (hunk 1 above)
Comment (Matioz) [reply modeler.py]: EnvExpert, EnvEstimator, EnvSimulator, Scout

Comment (piojanu) [reply modeler.py]: Hmm Scout brzmi nieźle, ale jak to uargumentujemy xD Że przeszukuje możliwe stany i uczy model? 😛

**Pull request merged**